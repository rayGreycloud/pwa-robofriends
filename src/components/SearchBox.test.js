import React from 'react';
import { shallow } from 'enzyme';

import SearchBox from './SearchBox';

describe('<SearchBox />', () => {
  it('should render component', () => {
    const mockSearchChange = jest.fn();
    expect(
      shallow(<SearchBox searchChange={mockSearchChange} />),
    ).toMatchSnapshot();
  });
});
