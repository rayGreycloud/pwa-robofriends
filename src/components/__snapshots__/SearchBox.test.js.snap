// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    searchChange={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        aria-label="Search Robots"
        className="pa3 ba b--green bg-lightest-blue"
        onChange={[Function]}
        placeholder="search robots"
        type="search"
/>,
      "className": "pa2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Search Robots",
        "className": "pa3 ba b--green bg-lightest-blue",
        "onChange": [Function],
        "placeholder": "search robots",
        "type": "search",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          aria-label="Search Robots"
          className="pa3 ba b--green bg-lightest-blue"
          onChange={[Function]}
          placeholder="search robots"
          type="search"
/>,
        "className": "pa2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Search Robots",
          "className": "pa3 ba b--green bg-lightest-blue",
          "onChange": [Function],
          "placeholder": "search robots",
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
